framework/fp_vectors/Floats.h

```diff
@@ -133,6 +158,166 @@ typedef SANDSTONE_FP16_TYPE fp16_t;
+    static constexpr BFloat8 max()      { return BFloat8(Holder{ 0b0'11110'11 }); }
+    static constexpr BFloat8 infinity() { return BFloat8(Holder{ 0b0'11111'00 }); }
+
+    constexpr inline bool is_negative() const { return (sign != 0); }
+    constexpr inline bool is_zero() const     { return (value == 0); }
+    constexpr inline bool is_denormal() const { return (exponent == BFLOAT8_DENORM_EXPONENT) && (mantissa != 0); }
+    constexpr inline bool is_valid() const    { return (exponent != BFLOAT8_NAN_EXPONENT); }
+    constexpr inline bool is_inf() const      { return (exponent == BFLOAT8_INFINITY_EXPONENT) && (mantissa == 0); }
+    constexpr inline bool is_overflow() const { return (exponent == BFLOAT8_EXPONENT_MASK) && (mantissa == BFLOAT8_OVERFLOW_MANTISSA); }
+    constexpr inline bool is_nan() const      { return (exponent == BFLOAT8_NAN_EXPONENT) && ((mantissa == BFLOAT8_SNAN_AT_INPUT_MANTISSA) || (mantissa == BFLOAT8_QNAN_AT_INPUT_MANTISSA)); }
+    constexpr inline bool is_snan() const     { return (exponent == BFLOAT8_NAN_EXPONENT) && (mantissa == BFLOAT8_SNAN_AT_INPUT_MANTISSA); }
+    constexpr inline bool is_qnan() const     { return (exponent == BFLOAT8_NAN_EXPONENT) && (mantissa == BFLOAT8_QNAN_AT_INPUT_MANTISSA); }
+
+    constexpr inline BFloat8 operator-() const {
+        return { (uint8_t) (sign ^ 1), exponent, mantissa };
+    }
+
+private:
+    struct Holder { uint8_t payload; };
+    explicit constexpr BFloat8(Holder h) : payload(h.payload) {}
+#endif
+};
+typedef struct BFloat8 BFloat8;
+
+// C interface
+static inline bool BFloat8_is_negative(BFloat8 f) { return (f.sign != 0); }
+static inline bool BFloat8_is_zero(BFloat8 f)     { return (f.value == 0); }
+static inline bool BFloat8_is_denormal(BFloat8 f) { return (f.exponent == BFLOAT8_DENORM_EXPONENT) && (f.mantissa != 0); }
+static inline bool BFloat8_is_valid(BFloat8 f)    { return (f.exponent != BFLOAT8_NAN_EXPONENT); }
+static inline bool BFloat8_is_inf(BFloat8 f)      { return (f.exponent == BFLOAT8_INFINITY_EXPONENT) && (f.mantissa == 0); }
+static inline bool BFloat8_is_overflow(BFloat8 f) { return (f.exponent == BFLOAT8_NAN_EXPONENT) && (f.mantissa == BFLOAT8_OVERFLOW_MANTISSA); }
+static inline bool BFloat8_is_nan(BFloat8 f)      { return (f.exponent == BFLOAT8_NAN_EXPONENT) && ((f.mantissa == BFLOAT8_SNAN_AT_INPUT_MANTISSA) || (f.mantissa == BFLOAT8_QNAN_AT_INPUT_MANTISSA)); }
+static inline bool BFloat8_is_snan(BFloat8 f)     { return (f.exponent == BFLOAT8_NAN_EXPONENT) && (f.mantissa == BFLOAT8_SNAN_AT_INPUT_MANTISSA); }
+static inline bool BFloat8_is_qnan(BFloat8 f)     { return (f.exponent == BFLOAT8_NAN_EXPONENT) && (f.mantissa == BFLOAT8_QNAN_AT_INPUT_MANTISSA); }
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+BFloat8 to_bfloat8_emulated(float f32);
+float from_bfloat8_emulated(BFloat8 f8);
+#ifdef __cplusplus
+}
+#endif
+
+static inline BFloat8 to_bfloat8(float f32) {
+    return to_bfloat8_emulated(f32);
```

Comment:
According "fresher" specs VCVT instructions exist for FP8 in DMR. Could we include these in a next commit?